tokio/src/runtime/basic_scheduler.rs

```diff
@@ -536,12 +536,17 @@ impl CoreGuard<'_> {
                         }
                     };
 
+                    let task_id = task.get_task_id();
+                    let old_id = context::set_current_task_id(Some(task_id));
+
                     let task = context.spawner.shared.owned.assert_owner(task);
 
                     let (c, _) = context.run_task(core, || {
                         task.run();
                     });
 
+                    context::set_current_task_id(old_id);
```

Comment:
we may want to do this via a `Drop` guard, rather than with a pair of function calls, to ensure that the previous task ID is always reset, even in the event of a panic in Tokio internals...